quicklens/src/main/scala-3/com.softwaremill.quicklens/package.scala

```diff
@@ -0,0 +1,288 @@
+      tree match {
+        case Select(deep, ident) =>
+          toPath(deep) :+ PathSymbol.Field(ident)
+        case Apply(Apply(Apply(TypeApply(Ident(s), typeTrees), idents), args), List(ident: Ident)) =>
+          idents.flatMap(toPath) :+ PathSymbol.FunctionDelegate(s, ident, typeTrees.last, args)
+        case a@Apply(Apply(TypeApply(Ident(s), typeTrees), idents), List(ident: Ident)) =>
+          idents.flatMap(toPath) :+ PathSymbol.FunctionDelegate(s, ident, typeTrees.last, List.empty)
+        case Apply(deep, idents) =>
+          toPath(deep) ++ idents.flatMap(toPath)
+        case i: Ident if i.name.startsWith("_") =>
+          Seq.empty
+        case _ =>
+          report.throwError(unsupportedShapeInfo(tree))
+      }
+    }
+
+    def termMethodByNameUnsafe(term: Term, name: String): Symbol = {
+      term.tpe.typeSymbol.memberMethod(name).head
+    }
+
+    def termAccessorMethodByNameUnsafe(term: Term, name: String): (Symbol, Int) = {
+      val caseFields = term.tpe.typeSymbol.caseFields
+      val idx = caseFields.map(_.name).indexOf(name)
+      (caseFields.find(_.name == name).get, idx+1)
+    }
+
+    def mapToCopy(mod: Expr[A => A], objTerm: Term, path: Seq[PathSymbol]): Term = path match
+      case Nil =>
+        val apply = termMethodByNameUnsafe(mod.asTerm, "apply")
+        Apply(Select(mod.asTerm, apply), List(objTerm))
+      case (field: PathSymbol.Field) :: tail =>
+        val copy = termMethodByNameUnsafe(objTerm, "copy")
+        val (fieldMethod, idx) = termAccessorMethodByNameUnsafe(objTerm, field.name)
+        val namedArg = NamedArg(field.name, mapToCopy(mod, Select(objTerm, fieldMethod), tail))
+        val fieldsIdxs = 1.to(objTerm.tpe.typeSymbol.caseFields.length)
+        val args = fieldsIdxs.map { i =>
+          if(i == idx) namedArg
+          else Select(objTerm, termMethodByNameUnsafe(objTerm, "copy$default$" + i.toString))
+        }.toList
+        Apply(
+          Select(objTerm, copy),
+          args
+        )
+      case (f: PathSymbol.FunctionDelegate) :: tail =>
+        val defdefSymbol = Symbol.newMethod(
+          Symbol.spliceOwner,
+          "$anonfun",
+          MethodType(List("x"))(_ => List(f.typeTree.tpe), _ => f.typeTree.tpe)
+        )
+        val fMethod = termMethodByNameUnsafe(f.givn, f.name)
+        val fun = TypeApply(
+          Select(f.givn, fMethod),
+          List(f.typeTree)
+        )
+        val defdefStatements = DefDef(
+          defdefSymbol, {
+            case List(List(x)) => Some(mapToCopy(mod, x.asExpr.asTerm, tail))
+          }
+        )
+        val closure = Closure(Ref(defdefSymbol), None)
+        val block = Block(List(defdefStatements), closure)
+        Apply(fun, List(objTerm, block) ++ f.args)
+    
+    val focusTree: Tree = focus.asTerm
+    val path = focusTree match {
+      case Inlined(_, _, Block(List(DefDef(_, _, _, Some(p))), _)) =>
+        toPath(p)
+      case Block(List(DefDef(_, _, _, Some(p))), _) =>
```

Comment:
same as above - I suppose one case is when we're using `obj.modify(path)`, the other when `modify(obj)(path)`?

Reply:
One case is for a single path and the other is for paths from modifyAll. Added a comment

Reply:
Ah ok, and the second isn't `Inlined` because it is unwrapped ealier using `Varargs` - thanks